Dockerfile

```diff
@@ -14,24 +14,27 @@ FROM base AS builder
 
 USER gen3
 
+# copy ONLY poetry artifact, install the dependencies but not the app;
+# this will make sure that the dependencies are cached
 COPY poetry.lock pyproject.toml /${appname}/
-
-RUN poetry install -vv --without dev --no-interaction
+RUN poetry install -vv --no-root --only main --no-interaction
 
 COPY --chown=gen3:gen3 . /${appname}
 COPY --chown=gen3:gen3 ./deployment/wsgi/wsgi.py /${appname}/wsgi.py
 
-# Run poetry again so this app itself gets installed too
+# install the app
 RUN poetry install --without dev --no-interaction
 
-RUN git config --global --add safe.directory /${appname} && COMMIT=`git rev-parse HEAD` && echo "COMMIT=\"${COMMIT}\"" > /${appname}/version_data.py \
-    && VERSION=`git describe --always --tags` && echo "VERSION=\"${VERSION}\"" >> /${appname}/version_data.py
+RUN git config --global --add safe.directory ${appname} && COMMIT=`git rev-parse HEAD` && echo "COMMIT=\"${COMMIT}\"" > ${appname}/version_data.py \
+    && VERSION=`git describe --always --tags` && echo "VERSION=\"${VERSION}\"" >> ${appname}/version_data.py
 
 # Final stage
 FROM base
 
 COPY --from=builder /${appname} /${appname}
 
+RUN dnf -y install vim
```

Comment:
I think having vim and nano in the base image would still be a better idea. Can approve this for now though!